Java/core/src/main/java/com/amazon/randomcutforest/imputation/Center.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.imputation;
+
+import static com.amazon.randomcutforest.CommonUtils.checkArgument;
+import static java.lang.Math.exp;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.BiFunction;
+
+/**
+ * the following class abstracts a single centroid representation of a group of
+ * points
+ */
+public class Center {
```

Comment:
Defined an interface ICluster -- it can enable other clustering algorithms; the use case in the imputation was to summarize a random conditional field. But clustering can be a standalone interest.